`test_temp_tables_cleanup_after_recovery` fails on files cleanup

Output of the test:

```
FAIL: test_temp_tables_cleanup_after_recovery (test.t.recovery_test.RecoveryTest.test_temp_tables_cleanup_after_recovery)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/orioledb/orioledb/orioledb/test/t/recovery_test.py", line 2129, in test_temp_tables_cleanup_after_recovery
    self.assertEqual(len(sorted(os.listdir(db_dir))), 0)
AssertionError: 4 != 0
```

--- a/include/catalog/o_tables.h
+++ b/include/catalog/o_tables.h
@@ -145,7 +145,7 @@ extern void o_tables_drop_all(OXid oxid, CommitSeqNo csn, Oid database_id);
 extern void o_tables_drop_columns_by_type(OXid oxid, CommitSeqNo csn, Oid type_oid);
 
 /* Drops all temporary tables that left after crash */
-extern void o_tables_drop_all_temporary(void);
+extern void o_tables_truncate_all_unlogged(void);
 
 /* Adds a new table to o_tables list */
 extern bool o_tables_add(OTable *table, OXid oxid, CommitSeqNo csn);

--- a/src/catalog/o_tables.c
+++ b/src/catalog/o_tables.c
@@ -98,7 +98,7 @@ typedef struct
 static void o_table_tupdesc_init_entry(TupleDesc desc, AttrNumber att_num, char *name, OTableField *field);
 static void o_tables_foreach_callback(ORelOids oids, void *arg);
 static void o_tables_drop_all_callback(ORelOids oids, void *arg);
-static void o_tables_drop_all_temporary_callback(OTable *o_table, void *arg);
+static void o_tables_truncate_unlogged_callback(OTable *o_table, void *arg);
 static void o_table_oids_array_callback(ORelOids oids, void *arg);
 static inline void o_tables_rel_fill_locktag(LOCKTAG *tag, ORelOids *oids, int lockmode, bool checkpoint);
 
@@ -1057,7 +1057,7 @@ o_tables_drop_all(OXid oxid, CommitSeqNo csn, Oid database_id)
 }
 
 void
-o_tables_drop_all_temporary()
+o_tables_truncate_all_unlogged()
 {
 	OTablesDropAllArg arg;
 	OXid		oxid;
@@ -1068,7 +1068,7 @@ o_tables_drop_all_temporary()
 	arg.oxid = oxid;
 	arg.csn = oSnapshot.csn;
 
-	o_tables_foreach(o_tables_drop_all_temporary_callback,
+	o_tables_foreach(o_tables_truncate_unlogged_callback,
 					 &o_non_deleted_snapshot, &arg);
 }
 
@@ -1504,60 +1504,9 @@ o_tables_drop_all_callback(ORelOids oids, void *arg)
 }
 
 static void
-o_tables_drop_all_temporary_callback(OTable *o_table, void *arg)
+o_tables_truncate_unlogged_callback(OTable *o_table, void *arg)
 {
-	OTablesDropAllArg *drop_arg = (OTablesDropAllArg *) arg;
-
-	if (o_table->persistence == RELPERSISTENCE_TEMP)
-	{
-		OTableChunkKey key;
-		bool		result;
-		OAutonomousTxState state;
-
-		key.oids = o_table->oids;
-		key.chunknum = 0;
-
-		start_autonomous_transaction(&state);
-
-		o_tables_table_meta_lock(NULL);
-		PG_TRY();
-		{
-			BTreeDescr *sys_tree;
-
-			o_tables_oids_indexes(o_table, NULL, get_current_oxid(), drop_arg->csn);
-			sys_tree = get_sys_tree(SYS_TREES_O_TABLES);
-			result = generic_toast_delete_optional_wal(&oTablesToastAPI,
-													   (Pointer) &key,
-													   get_current_oxid(),
-													   drop_arg->csn,
-													   sys_tree,
-													   false);
-			if (result)
-			{
-				ORelOids   *treeOids;
-				int			numTreeOids;
-				int			i;
-
-				treeOids = o_table_make_index_oids(o_table, &numTreeOids);
-				for (i = 0; i < numTreeOids; i++)
-				{
-					cleanup_btree(treeOids[i].datoid, treeOids[i].relnode,
-								  true);
-				}
-				pfree(treeOids);
-			}
-		}
-		PG_CATCH();
-		{
-			o_tables_table_meta_unlock(NULL, InvalidOid);
-			abort_autonomous_transaction(&state);
-			PG_RE_THROW();
-		}
-		PG_END_TRY();
-		o_tables_table_meta_unlock(NULL, InvalidOid);
-		finish_autonomous_transaction(&state);
-	}
-	else if (o_table->persistence == RELPERSISTENCE_UNLOGGED)
+	if (o_table->persistence == RELPERSISTENCE_UNLOGGED)
 	{
 		o_truncate_table(o_table->oids);
 		AcceptInvalidationMessages();

--- a/src/transam/undo.c
+++ b/src/transam/undo.c
@@ -2064,7 +2064,7 @@ orioledb_snapshot_hook(Snapshot snapshot)
 	if (*was_in_recovery &&
 		!pg_atomic_exchange_u32(after_recovery_cleaned, true))
 	{
-		o_tables_drop_all_temporary();
+		o_tables_truncate_all_unlogged();
 	}
 
 	for (i = 0; i < (int) UndoLogsCount; i++)

--- a/test/t/recovery_test.py
+++ b/test/t/recovery_test.py
@@ -2073,6 +2073,9 @@ class RecoveryTest(BaseTest):
 		node = self.node
 		node.append_conf('orioledb.recovery_pool_size = 1')
 		node.append_conf('orioledb.recovery_idx_pool_size = 1')
+		node.append_conf('autovacuum_naptime = 1s')
+		node.append_conf('autovacuum_vacuum_threshold = 0')
+		node.append_conf('autovacuum_vacuum_scale_factor = 0')
 		node.start()
 
 		node.safe_psql("""
@@ -2123,14 +2126,14 @@ class RecoveryTest(BaseTest):
 		file_num = 3  # PK trees
 		file_num += 1  # Indexes
 		db_dir = f"{node.data_dir}/orioledb_data/{cur_database}"
+
 		self.assertEqual(len(sorted(os.listdir(db_dir))), file_num)
 		node.start()
-		node.stop(['-m', 'immediate'])
-		self.assertEqual(len(sorted(os.listdir(db_dir))), 0)
-		node.start()
-		node.stop(['-m', 'immediate'])
 
-		node.start()
+		node.poll_query_until(
+		    "SELECT count(*) = 0 FROM pg_class WHERE relname LIKE 'o_test_%%'")
+		self.assertEqual(len(sorted(os.listdir(db_dir))), 0)
+
 		self.assertEqual(
 		    node.execute("""
 								SELECT c.relname

--- a/test/t/replication_test.py
+++ b/test/t/replication_test.py
@@ -812,6 +812,10 @@ class ReplicationTest(BaseTest):
 		node = self.node
 		node.append_conf('orioledb.recovery_pool_size = 1')
 		node.append_conf('orioledb.recovery_idx_pool_size = 1')
+		node.append_conf('autovacuum_naptime = 1s')
+		node.append_conf('autovacuum_vacuum_threshold = 0')
+		node.append_conf('autovacuum_vacuum_scale_factor = 0')
+
 		node.start()
 		with self.node as master:
 			master.safe_psql("""
@@ -893,15 +897,11 @@ class ReplicationTest(BaseTest):
 
 					master.stop(['-m', 'immediate'])
 					master.start()
-					master.stop(['-m', 'immediate'])
 
-					master.start()
-					master.stop()
-					master.start()
-					master.stop()
-					master.start()
-					master.stop()
-					master.start()
+					master.poll_query_until(
+					    "SELECT count(*) = 0 FROM pg_class WHERE relname LIKE 'o_test%%'"
+					)
+
 					self.assertEqual(
 					    master.execute("""
 											SELECT c.relname

